js/components/generate/generate.js

```diff
@@ -236,9 +248,34 @@ class GenerateForm extends React.Component {
       })
   }
 
+  /**
+   * Produce an appropriate warning message string in the case that no courses have been entered
+   * or that one or more invalid courses have been entered.
+   * @param {string} coursesString - The string version of the inputted courses
+   * @param {string[]} invalidCourses - The array of invalid course codes
+   * @returns {string} The warning message string.
+   */
+  computeMessage(coursesString, invalidCourses) {
+    if (!coursesString.length) {
+      return "Cannot generate graph -- no courses entered!"
```

Comment:
Replace the `--` with an endash, –. (*Warning*: copy that character directly, don't try to type it in yourself. It's different than the regular hyphen that appears on your keyboard.)